holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
``element._properties`` sounds like a reasonable suggestion. For the time being ``_integer_indexable_1d`` isn't really correct though, it only adds an integer index if you don't supply x-values yourself, ``_add_1d_index`` maybe?